pkg/testing/functional.go

```diff
@@ -132,10 +132,11 @@ func WithNoTraffic(reason, message string) PodAutoscalerOption {
 	}
 }
 
-// WithPADeletionTimestamp will set the DeletionTimestamp on the PodAutoscaler.
-func WithPADeletionTimestamp(r *autoscalingv1alpha1.PodAutoscaler) {
+// WithDeletionTimestamp will set the DeletionTimestamp on the object.
+func WithDeletionTimestamp[T metav1.Object](obj T) T {
```

Comment:
amazing - there's probably a ton of cleanup we can do in a lot of these tests

Reply:
Maybe, I had not checked, only stumbled over it when I searched for a modifier for my use case. I assume you only commented this without expecting that I now do this for the full code base of this repo?

Reply:
> I assume you only commented this without expecting that I now do this for the full code base of this repo?

yeah exactly - just an observation